supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesInitializationActor.scala

```diff
@@ -0,0 +1,141 @@
+  val MemoryKey = "memory"
+  val MemoryDefaultValue = "2 GB"
+  val DisksKey = "disks"
+  val DisksDefaultValue = s"${JesWorkingDisk.Name} 10 SSD"
+  val DefaultJesWorkingDisk = JesAttachedDisk.parse(DisksDefaultValue).get
+  val FailOnStderrDefaultValue = false
+  val ContinueOnRCDefaultValue = 0
+
+  def props(workflowDescriptor: BackendWorkflowDescriptor, calls: Seq[Call], configurationDescriptor: BackendConfigurationDescriptor): Props =
+    Props(new JesInitializationActor(workflowDescriptor, calls, configurationDescriptor))
+}
+
+class JesInitializationActor(override val workflowDescriptor: BackendWorkflowDescriptor,
+                             override val calls: Seq[Call],
+                             override val configurationDescriptor: BackendConfigurationDescriptor) extends BackendWorkflowInitializationActor {
+  /**
+    * Abort all initializations.
+    */
+  override def abortInitialization(): Future[WorkflowAbortResponse] = ???
+
+  //TODO: Workflow options may need to be validated for JES.
+  //TODO: Move specifics from Jes to JesBackendInitializationActor.
+
+  /**
+    * A call which happens before anything else runs
+    */
+  override def beforeAll(): Future[Unit] = {
+    Future.successful {
+      //Add missing code here.
+    }
+  }
+
+  /**
+    * Validates runtime attributes for one specific call.
+    *
+    * @param runtimeAttributes Runtime Attributes with already evaluated values.
+    * @return If all entries from runtime attributes section are valid Success otherwise
+    *         Failure with the aggregation of errors.
+    */
+  override def validateRuntimeAttributes(runtimeAttributes: EvaluatedRuntimeAttributes): ErrorOr[(Int, Vector[String], Int, Int, Double, Seq[JesAttachedDisk], Option[String], Boolean, ContinueOnReturnCode)] = {
+    val cpu = validateCpu(runtimeAttributes.get(CpuKey))
+    val zones = validateZone(runtimeAttributes.get(ZonesKey))
+    val preemptible = validatePreemptible(runtimeAttributes.get(PreemptibleKey))
+    val bootDiskSize = validateBootDisk(runtimeAttributes.get(BootDiskSizeKey))
+    val memory = RuntimeAttributes.validateMemoryValue(runtimeAttributes.getOrElse(MemoryKey, WdlString(MemoryDefaultValue))) match {
+      case scala.util.Success(x) => x.to(MemoryUnit.GB).amount.successNel
+      case scala.util.Failure(x) => x.getMessage.failureNel
+    }
+    val disks = validateLocalDisks(runtimeAttributes.get(DisksKey))
+    val docker = validateDocker(runtimeAttributes.get(Docker), () => None.successNel)
+    val failOnStderr = validateFailOnStderr(runtimeAttributes.get(FailOnStderr), () => FailOnStderrDefaultValue.successNel)
+    val continueOnReturnCode = validateContinueOnReturnCode(runtimeAttributes.get(ContinueOnReturnCode),
+      () => ContinueOnReturnCodeSet(Set(ContinueOnRCDefaultValue)).successNel)
+    (cpu |@| zones |@| preemptible |@| bootDiskSize |@| memory |@| disks |@| docker |@| failOnStderr |@| continueOnReturnCode) {
+      (_, _, _, _, _, _, _, _, _)
```

Comment:
It'd be nicer for backend implementors if they just return a List[ErrorOr[Unit]] and this ugly scalaz `fold` is done in the `BackendInitializationActor` trait